plugin/trino-druid/src/test/java/io/trino/plugin/druid/DruidSqlDataTypeTest.java

```diff
@@ -0,0 +1,180 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.druid;
+
+import io.trino.Session;
+import io.trino.spi.type.Type;
+import io.trino.sql.query.QueryAssertions;
+import io.trino.sql.query.QueryAssertions.QueryAssert;
+import io.trino.testing.MaterializedResult;
+import io.trino.testing.QueryRunner;
+import io.trino.testing.datatype.ColumnSetup;
+import io.trino.testing.datatype.DataSetup;
+import io.trino.testing.sql.TestTable;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.IntStream;
+
+import static com.google.common.base.Preconditions.checkState;
+import static java.lang.String.format;
+import static java.util.Collections.unmodifiableList;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.joining;
+import static org.assertj.core.api.Assertions.assertThat;
+
+public final class DruidSqlDataTypeTest
```

Comment:
What would it take to avoid copying `SqlDataTypeTest`?

Reply:
`SqlDataTypeTest` relies on `TestTable`, in whose constructor CREATE TABLE and INSERT DATA will be run -- this is not suitable for Druid.
https://github.com/trinodb/trino/issues/8404#issuecomment-878830576

Reply:
There are 3 things in the game: `SqlDataTypeTest`, `DataSetup` and `TestTable`

- `SqlDataTypeTest` delegates all the setup  to `DataSetup`, so only this needs to be specific for Druid
- `TestTable` is used in `DataSetup` interface
- `SqlDataTypeTest` uses `TestTable` for two things:
   - knowing table name
   - removing table with `TestTable#close` (cleanup)

We should address this by introducing a new interface:

- call it `TemporaryRelation`
- with methods `getRelationName` and `close()` (extending from `AutoCloseable`)
- make `TestTable` implement `TemporaryRelation`
- use it in `DataSetup` interface definition
- all current `DataSetup` implementations will continue to use `TestTable`
- you will write a new implementation for Druid.

Would that work?

cc @hashhar